src/JMS/Serializer/Accessor/Updater/ClassAccessorUpdater.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace JMS\Serializer\Accessor\Updater;
+
+use JMS\Serializer\Exception\RuntimeException;
+use JMS\Serializer\Metadata\ClassMetadata;
+use JMS\Serializer\Metadata\ClassMetadataUpdaterInterface;
+use JMS\Serializer\Metadata\PropertyMetadata;
+
+class ClassAccessorUpdater implements ClassMetadataUpdaterInterface
+{
+    /**
+     * @var string
+     */
+    private $defaultType;
+
+    /**
+     * @var string
+     */
+    private $defaultNaming;
+
+    /**
+     * @var array
+     */
+    private $getterPrefixes;
+
+    /**
+     * @var array
+     */
+    private $setterPrefixes;
+
+    public function __construct(
+        $defaultType = PropertyMetadata::ACCESS_TYPE_PROPERTY,
+        $defaultNaming = PropertyMetadata::ACCESS_TYPE_NAMING_EXACT,
+        array $getterPrefixes = ['get', 'is', 'has'],
+        array $setterPrefixes = ['set', 'update']
+    )
+    {
+        $this->defaultType = $defaultType;
+        $this->defaultNaming = $defaultNaming;
+        $this->getterPrefixes = $getterPrefixes;
+        $this->setterPrefixes = $setterPrefixes;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function update(ClassMetadata $classMetadata)
+    {
+        foreach ($classMetadata->propertyMetadata as $propertyMetadata) {
+            $this->updateAccessors($propertyMetadata, $classMetadata);
+        }
+    }
+
+    /**
+     * @param string $accessorPrefix
+     * @param string $propertyName
+     * @param string $naming
+     *
+     * @return string|null
+     */
+    protected function getAccessorName($accessorPrefix, $propertyName, $naming)
```

Comment:
This method is totally decoupled from the class logic, so it could be refactored out making easy extending list of supported naming strategies